src/content/en/2021/performance.md

```diff
@@ -1,20 +1,532 @@
+
+### Cumulative Layout Shift (CLS)
+
+{{ figure_markup(
+  image="performance-CLS-by-device.png",
+  caption="CLS performance by device",
+  description="Stacked bar chart showing CLS performance for desktop and mobile devices. For desktop devices, 62% of websites had good (< 0.1), 23% needs improvement, and 15% poor (>= 0.25) CLS performance. For mobile, 62% were good, 21% needs improvement, and 17% poor.",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vQj6-t49BNV_5W-w83AABkUoo6klfyUyIz13yKShLAzK8qGs5lJ9TKcggIEp6JgxikVF-UJBAHpsrNl/pubchart?oid=160840238&format=interactive",
+  sheets_gid="730149797",
+  sql_file="web_vitals_by_device.sql"
+  )
+}}
+
+<a hreflang="en" href="https://web.dev/cls/">Cumulative Layout Shift (CLS)</a> is characterized by how much shift a user experiences, not how long it takes to visually see something like FCP and LCP. As such, performance by device was fairly equivalent.
+
+{{ figure_markup(
+  image="performance-CLS-by-ect.png",
+  caption="CLS performance by connection type",
+  description="Stacked bar chart showing CLS performance for effective connection type. For offline websites, 87% had good and 4% poor performance. For slow 2G and 2G, 53-52% were good, 15% needs improvement, and 32-33% poor. For 3G, 58% were good, 16% needs improvement, and 26% poor. For 4G, 69% were good, 13% needs improvement, and 18% poor.",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vQj6-t49BNV_5W-w83AABkUoo6klfyUyIz13yKShLAzK8qGs5lJ9TKcggIEp6JgxikVF-UJBAHpsrNl/pubchart?oid=1650043738&format=interactive",
+  sheets_gid="730149797",
+  sql_file="web_vitals_by_eff_connection_type.sql"
+  )
+}}
+
+Performance degradation from 4G to 3G and below was not as pronounced as with FCP and LCP. Some degradation exists, but it's not reflected in the device data, only the connection type.
+
+Offline websites had the highest CLS performance of all connection types. For sites with service worker caching, some assets like images and ads that would otherwise cause layout shifts may not be cached. Thus, they would never load and never cause a layout shift. Often fallback HTML for these sites can be more basic versions of the online website.
+
+{{ figure_markup(
+  image="performance-CLS-by-rank.png",
+  caption="CLS performance by rank",
+  description="Stacked bar chart showing CLS performance for each ranking group. For the top 1,000 websites, 53% had good, 25% needs improvement, and 21% poor performance. For the top 10,000, 46% were good, 27% needs improvement, and 27% poor. For the top 100,000, 48% were good, 26% needs improvement, and 26% poor. For the top 1,000,000, 54% were good, 25% needs improvement, and 21% poor. For all, 61% were good, 23% needs improvement, and 16% poor.",
```

Comment:
inconsistent between `good` and `poor`

Reply:
Maybe you want the verb for every category? I figured that would make the screen reader time even longer. For "needs improvement" and "poor", both are bucket names so was just trying to get the numbers across quickly and only provide the verb for the first.

Reply:
If I'm understanding correctly, that would change the description for this and all similar to something like:

> Stacked bar chart showing CLS performance for each ranking group. For the top 1,000 websites, 53% had good, 25% had needs improvement, and 21% had poor performance. For the top 10,000, 46% were good, 27% were needs improvement, and 27% were poor. For the top 100,000, 48% were good, 26% were needs improvement, and 26% were poor. For the top 1,000,000, 54% were good, 25% were needs improvement, and 21% were poor. For all, 61% were good, 23% were needs improvement, and 16% were poor.

Reply:
You can also remove `had`?  I just meant you had one part of the list with the verb and the other parts of the list without.  I think if I were being strictly grammatical, I'd need the verb in there?